backend/clickhouse/query.go

```diff
@@ -1051,6 +1283,165 @@ func (client *Client) getSamplingStats(ctx context.Context, tables []string, pro
 	return statsByTable, nil
 }
 
+func (client *Client) readMetricsSql(ctx context.Context, input ReadMetricsInput, config model.TableConfig) (*modelInputs.MetricsBuckets, error) {
+	builderInfo, err := builderFromSql(config, input)
+	if err != nil {
+		return nil, err
+	}
+	applyBlockFilter(builderInfo.Builder, input)
+	addAttributes(config, builderInfo.Fields, input.ProjectIDs, input.Params, builderInfo.Builder, true)
+
+	sql, args := builderInfo.Builder.BuildWithFlavor(sqlbuilder.ClickHouse)
+
+	sql += fmt.Sprintf(" SETTINGS SQL_highlight_project_id=%d", input.ProjectIDs[0])
```

Comment:
just checking, but because this comes last, even if the user sets this in the query this would be overriden?

Reply:
in the end, I'm doing 3 things:
- adding `SQL_highlight_project_id` setting via the context
- making sure `SETTINGS` clauses are parsed and then ignored / not included in the result query
- returning an error for any query containing `SQL_highlight_project_id` - the `SETTINGS` clause only allows literals so as far as I can tell it's not possible to construct this dynamically (e.g. `concat('SQL_highli','ght_project_id')